mono/metadata/marshal-ilgen.c

```diff
@@ -1171,7 +1171,8 @@ emit_thread_interrupt_checkpoint_call (MonoMethodBuilder *mb, gpointer checkpoin
 	/* Throw the exception returned by the checkpoint function, if any */
 	mono_mb_emit_byte (mb, CEE_DUP);
 	pos_noex = mono_mb_emit_branch (mb, CEE_BRFALSE);
-	mono_mb_emit_byte (mb, CEE_THROW);
+	mono_mb_emit_icall (mb, mono_marshal_rethrow_exception);
```

Comment:
Is it okay to rethrow an exception that was never thrown?  Some of the code paths in `mono_thread_execute_interruption` return brand-new interruptions (e.g. `mono_exception_new_thread_interrupted` in the `thread_interrupt_requested` case).

In other cases `ThreadState_AbortRequested` we re-fly an existing exception (I can't quite tell if the same abort exception gets used multiple times - I think it's possible with some creative use of `ResetAbort`).

Reply:
The `mini` helper in question is used to resume throwing exceptions that were already taken out of the in-flight status. 

This code path seemed very hard to actually hit when testing. The other jit icall I added takes 99.9% of the rethrows. Do you think this is more llvm-specific? Some comments might be in order if that's the case.

Reply:
I don't think it's LLVM specific.  probably since we AOT corlib under normal circumstances you'd only hit this in the early bootstrapping profiles or for the handful of icalls that aren't in corlib.